run_unittests.py

```diff
@@ -2712,6 +2713,29 @@ def test_old_gnome_module_codepaths(self):
         self.build()
         mesonbuild.modules.gnome.native_glib_version = None
 
+    @unittest.skipIf(shutil.which('pkg-config') is None, 'Pkg-config not found.')
+    def test_pkgconfig_usage(self):
+        testdir1 = os.path.join(self.unit_test_dir, '24 pkgconfig usage/dependency')
+        testdir2 = os.path.join(self.unit_test_dir, '24 pkgconfig usage/dependee')
+        with tempfile.TemporaryDirectory() as tempdirname:
+            self.init(testdir1, ['--prefix=' + tempdirname, '--libdir=lib'], default_args=False)
+            self.install(use_destdir=False)
+            shutil.rmtree(self.builddir)
+            os.mkdir(self.builddir)
+            pkg_dir = os.path.join(tempdirname, 'lib/pkgconfig')
+            self.assertTrue(os.path.exists(os.path.join(pkg_dir, 'libpkgdep.pc')))
+            lib_dir = os.path.join(tempdirname, 'lib')
+            os.environ['PKG_CONFIG_PATH'] = pkg_dir
+            self.init(testdir2)
+            self.build()
+            myenv = os.environ.copy()
+            myenv['LD_LIBRARY_PATH'] = lib_dir
+            self.assertTrue(os.path.isdir(lib_dir))
+            self.assertTrue(os.path.isfile(os.path.join(lib_dir, 'libpkgdep.so')))
+            test_exe = os.path.join(self.builddir, 'pkguser')
+            self.assertTrue(os.path.isfile(test_exe))
+            subprocess.check_call(test_exe, env=myenv)
+
```

Comment:
test_pkgconfig_gen_deps() already pretty much does the same thing. I think it would be better to add test cases there instead of a completely new test case. Also I added self.new_builddir() for this kind of test.